packages/valory/skills/score_read_abci/rounds.py

```diff
@@ -103,7 +103,16 @@ def end_block(self) -> Optional[Tuple[BaseSynchronizedData, Event]]:
             wallet_to_users = cast(
                 SynchronizedData, self.synchronized_data
             ).wallet_to_users
-            wallet_to_users.update(payload["wallet_to_users"])
+
+            for wallet, user in payload["wallet_to_users"].items():
+                if user in wallet_to_users.values():
+                    # Delete the old entry
+                    old_wallet = list(wallet_to_users.keys())[
+                        list(wallet_to_users.values()).index(user)
+                    ]
+                    del wallet_to_users[old_wallet]
+                # Add the new entry
+                wallet_to_users[wallet] = user
```

Comment:
We need to check wether a user already existed in our db and update the entry in that case.